pacakge `gimpPlugins.gmic` failed to build

### Steps To Reproduce
Steps to reproduce the behavior:
1. build `gimpPlugins.gmic`
2. `nix build nixpkgs/nixos-unstable#gimpPlugins.gmic`

### Build log
```
error: builder for '/nix/store/d0y9vpmxhb6xwqv6d3zirkxb7hss0rhs-gmic-qt-gimp-3.1.6.drv' failed with exit code 1;
       last 10 log lines:
       >     CMAKE_INSTALL_OLDINCLUDEDIR
       >     CMAKE_INSTALL_SBINDIR
       >     CMAKE_POLICY_DEFAULT_CMP0025
       >
       > 
       > -- Build files have been written to: /build/source/build
       > cmake: enabled parallel building
       > building
       > build flags: -j8
       > ninja: error: '/build/source/translations/filters/fr.qm', needed by 'gmic_gimp_qt_autogen/EWIEGA46WW/qrc_wip_translations.cpp', missing and no known rule to make it
```

Full log: https://gist.github.com/ShivanshuKantPrasad/87dbc749dc97d75eacb7db68f0837408

### Additional context
Package built successfully in #209831.
Package failed in #211445.
I assume package failed to built somewhere in between these two pull request.

### Notify maintainers
@jtojnar

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.86, NixOS, 23.05 (Stoat), 23.05.20230115.6dccdc4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.12.0`
 - channels(shivanshu): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

@jtojnar, tried but had a compilation issue:

```
diff --git a/pkgs/tools/graphics/gmic-qt/default.nix b/pkgs/tools/graphics/gmic-qt/default.nix
index a937ea51784..0e6d0b0adb4 100644
--- a/pkgs/tools/graphics/gmic-qt/default.nix
+++ b/pkgs/tools/graphics/gmic-qt/default.nix
@@ -2,7 +2,6 @@
 , mkDerivation
 , variant ? "standalone"
 , fetchFromGitHub
-, fetchpatch
 , cmake
 , pkg-config
 , ninja
@@ -59,13 +58,13 @@ assert lib.assertMsg (builtins.all (d: d != null) variants.${variant}.extraDeps
 
 mkDerivation rec {
   pname = "gmic-qt${lib.optionalString (variant != "standalone") "-${variant}"}";
-  version = "3.1.6";
+  version = "3.2.0";
 
   src = fetchFromGitHub {
     owner = "c-koi";
     repo = "gmic-qt";
     rev = "v.${version}";
-    sha256 = "sha256-/5wDHvJSMgEheg8YV4W40wUiHz25emIoFnGdfO8i92g=";
+    sha256 = "sha256-I5XC7zbDyBPFj4zul9rshoyeVV0hRQQ3aZQzEvYrgdc=";
   };
 
   nativeBuildInputs = [
@@ -95,15 +94,6 @@ mkDerivation rec {
     "-DENABLE_SYSTEM_GMIC:BOOL=ON"
   ];
 
-  patches = [
-    # NOTE: this should be removed when a new version is released.
-    (fetchpatch {
-      name = "fix_filter_translation_scripts.patch";
-      url = "https://github.com/c-koi/gmic-qt/commit/ccb9f29eda239d0c80663593cd90a6548c935b39.patch";
-      sha256 = "sha256-OzuJ6yGuDJweQ+1uin/pmJqZV79bN9E1Zuo+0iciwzg=";
-    })
-  ];
-
   postPatch = ''
     patchShebangs \
       translations/filters/csv2ts.sh \
```

Error:

```
[61/96] Building CXX object CMakeFiles/gmic_qt.dir/src/HeadlessProcessor.cpp.o
FAILED: CMakeFiles/gmic_qt.dir/src/HeadlessProcessor.cpp.o 
/nix/store/h4xripzh81wc3wizprjiypfj7sjcj73j-gcc-wrapper-12.2.0/bin/g++ -DGMIC_HOST=standalone -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_NO_DEBUG_OUTPUT -DQT_WIDGETS_LIB -D_IS_LINUX_ -D_IS_UNIX_ -Dcimg_appname=\"gmic\" -Dcimg_display=1 -Dcimg_use_abort -Dcimg_use_cpp11=1 -Dcimg_use_curl -Dcimg_use_fftw3 -Dcimg_use_openmp -Dcimg_use_png -Dcimg_use_vt100 -Dcimg_use_zlib -Dgmic_community -Dgmic_gui -Dgmic_is_parallel -I/build/source/build -I/build/source -I/build/source/build/gmic_qt_autogen/include -I/build/source/src -isystem /nix/store/s03hssb0lv69357s1nd00pdpf9yf4xbf-qtbase-5.15.8-dev/include/QtCore -isystem /nix/store/s03hssb0lv69357s1nd00pdpf9yf4xbf-qtbase-5.15.8-dev/./mkspecs/linux-g++ -isystem /nix/store/s03hssb0lv69357s1nd00pdpf9yf4xbf-qtbase-5.15.8-dev/include/QtWidgets -isystem /nix/store/s03hssb0lv69357s1nd00pdpf9yf4xbf-qtbase-5.15.8-dev/include/QtGui -isystem /nix/store/s03hssb0lv69357s1nd00pdpf9yf4xbf-qtbase-5.15.8-dev/include/QtNetwork -DNDEBUG -Ofast -fopenmp -fPIC -std=gnu++11 -MD -MT CMakeFiles/gmic_qt.dir/src/HeadlessProcessor.cpp.o -MF CMakeFiles/gmic_qt.dir/src/HeadlessProcessor.cpp.o.d -o CMakeFiles/gmic_qt.dir/src/HeadlessProcessor.cpp.o -c /build/source/src/HeadlessProcessor.cpp
/build/source/src/HeadlessProcessor.cpp: In member function 'void GmicQt::HeadlessProcessor::startProcessing()':
/build/source/src/HeadlessProcessor.cpp:154:22: error: no matching function for call to 'gmic_library::gmic_list<float>::assign()'
  154 |   _gmicImages->assign();
      |   ~~~~~~~~~~~~~~~~~~~^~
In file included from /build/source/src/HeadlessProcessor.cpp:43:
/nix/store/921zg9g89yjrb7q34bmja7gcil01a5y6-gmic-3.2.0-dev/include/gmic.h:151:19: note: candidate: 'gmic_library::gmic_list<T>& gmic_library::gmic_list<T>::assign(unsigned int) [with T = float]'
  151 |     gmic_list<T>& assign(const unsigned int n);
      |                   ^~~~~~
/nix/store/921zg9g89yjrb7q34bmja7gcil01a5y6-gmic-3.2.0-dev/include/gmic.h:151:19: note:   candidate expects 1 argument, 0 provided
ninja: build stopped: subcommand failed.
```

--- a/pkgs/tools/graphics/gmic-qt/default.nix
+++ b/pkgs/tools/graphics/gmic-qt/default.nix
@@ -1,7 +1,8 @@
 { lib
 , mkDerivation
 , variant ? "standalone"
-, fetchFromGitHub
+, fetchzip
+, fetchpatch
 , cmake
 , pkg-config
 , ninja
@@ -16,7 +17,6 @@
 , curl
 , gimp ? null
 , gmic
-, cimg
 , qtbase
 , qttools
 , writeShellScript
@@ -54,12 +54,20 @@ mkDerivation rec {
   pname = "gmic-qt${lib.optionalString (variant != "standalone") "-${variant}"}";
   version = "3.2.1";
 
-  src = fetchFromGitHub {
-    owner = "c-koi";
-    repo = "gmic-qt";
-    rev = "v.${version}";
-    sha256 = "sha256-z+GtYLBcHVufXwdeSd8WKmPmU1+/EKMv26kNaEgyt5w=";
+  src = fetchzip {
+    url = "https://gmic.eu/files/source/gmic_${version}.tar.gz";
+    hash = "sha256-2lMnn19FcFKnfIjSxOObqxIjqLMUoWgi0ADZBCBePY4=";
   };
+
+  patches = [
+    (fetchpatch {
+      name = "gmic-qt-3.2.1-fix-system-gmic.patch";
+      url = "https://github.com/c-koi/gmic-qt/commit/e8d7a3523753ff592da63b1d54edf0921c54fe53.patch";
+      hash = "sha256-kBFZo2qvod4pH3oK8gvnmw39x6eMH9zjr4mMcY74mFo=";
+    })
+  ];
+
+  sourceRoot = "source/gmic-qt";
 
   nativeBuildInputs = [
     cmake
@@ -69,7 +77,6 @@ mkDerivation rec {
 
   buildInputs = [
     gmic
-    cimg
     qtbase
     qttools
     fftw
@@ -85,7 +92,8 @@ mkDerivation rec {
 
   cmakeFlags = [
     "-DGMIC_QT_HOST=${if variant == "standalone" then "none" else variant}"
-    "-DENABLE_SYSTEM_GMIC:BOOL=ON"
+    "-DENABLE_SYSTEM_GMIC=ON"
+    "-DENABLE_DYNAMIC_LINKING=ON"
   ];
 
   postPatch = ''
@@ -108,8 +116,6 @@ mkDerivation rec {
   };
 
   meta = with lib; {
-    # Broken since 3.2.0 update, cannot handle system gmic and cimg.
-    broken = true;
     description = variants.${variant}.description;
     homepage = "http://gmic.eu/";
     license = licenses.gpl3Plus;

--- a/pkgs/tools/graphics/gmic/default.nix
+++ b/pkgs/tools/graphics/gmic/default.nix
@@ -1,6 +1,7 @@
 { stdenv
 , lib
 , fetchFromGitHub
+, fetchpatch
 , fetchurl
 , cmake
 , ninja
@@ -14,6 +15,7 @@
 , libjpeg
 , libtiff
 , libpng
+, libX11
 , writeShellScript
 , common-updater-scripts
 , curl
@@ -37,6 +39,14 @@ stdenv.mkDerivation rec {
     hash = "sha256-oEH4GlSV+642TGSJJhV4yzydh1hAQZfzwaiPAZFNQtI=";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "gmic-3.2.1-fix-system-gmic.patch";
+      url = "https://github.com/GreycLab/gmic/commit/1fc184b40b7c31e7b100722d32cb9f7c5a79c90f.patch";
+      hash = "sha256-xLlZ87QUiBrfioG7DNXf0HioxqOu6HX+57LW2FGdbLI=";
+    })
+  ];
+
   # TODO: build this from source
   # https://github.com/dtschump/gmic/blob/b36b2428db5926af5eea5454f822f369c2d9907e/src/Makefile#L675-L729
   gmic_stdlib = fetchurl {
@@ -58,6 +68,7 @@ stdenv.mkDerivation rec {
     libjpeg
     libtiff
     libpng
+    libX11
     opencv
     openexr
     graphicsmagick

--- a/pkgs/tools/graphics/gmic/default.nix
+++ b/pkgs/tools/graphics/gmic/default.nix
@@ -15,7 +15,6 @@
 , libjpeg
 , libtiff
 , libpng
-, libX11
 , writeShellScript
 , common-updater-scripts
 , curl
@@ -42,8 +41,8 @@ stdenv.mkDerivation rec {
   patches = [
     (fetchpatch {
       name = "gmic-3.2.1-fix-system-gmic.patch";
-      url = "https://github.com/GreycLab/gmic/commit/1fc184b40b7c31e7b100722d32cb9f7c5a79c90f.patch";
-      hash = "sha256-xLlZ87QUiBrfioG7DNXf0HioxqOu6HX+57LW2FGdbLI=";
+      url = "https://github.com/GreycLab/gmic/commit/9db3f6a39d9ed67b4279654da88993a8057575ff.patch";
+      hash = "sha256-JznKCs56t6cJ4HLqlhMZjSOupEB8cdkn3j6RgZpcpzo=";
     })
   ];
 
@@ -68,7 +67,6 @@ stdenv.mkDerivation rec {
     libjpeg
     libtiff
     libpng
-    libX11
     opencv
     openexr
     graphicsmagick

